slo-libsonnet/latency-burn.libsonnet

```
@@ -0,0 +1,160 @@
+local util = import '_util.libsonnet';
+{
+  local sumRule(param) = {
+    local slo = {
+      metric: error 'must set metric for sum recording',
+      selectors: error 'must set selectors for sum recording',
+      rate: error 'must set rate for sum recording',
+    } + param,
+
+    local labels =
+      util.selectorsToLabels(slo.selectors),
+
+    local recordingrule = {
+      expr: |||
+        sum(rate(%s{%s}[%s]))
+      ||| % [
+        slo.metric,
+        std.join(',', slo.selectors),
```

These slo.selectors need to be joined with the le="1.5" latencyTarget. Otherwise, we sum up all latency buckets... :grin: 
As this is just an array you should be able to have this as something along the lines of:
```suggestion
        std.join(',', slo.selectors + ['le="%s"' % slo.latencyTarget]),
```

---

yep , updated and this simplified the whole thing quite a bit.